ocis-pkg/log/log_test.go

```diff
@@ -0,0 +1,62 @@
+package log_test
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/onsi/gomega"
+
+	"github.com/owncloud/ocis/v2/ocis-pkg/log"
+)
+
+func TestDefault(t *testing.T) {
+	if os.Getenv("TestDefault") == "true" {
+		log.Default().Info().Msg("this is a test")
+		return
+	}
+
+	g := gomega.NewWithT(t)
+
+	tests := []struct {
+		name     string
+		args     []string
+		validate func(result string)
+	}{
+		{
+			name: "default",
+			args: []string{"OCIS_LOG_PRETTY=false", "OCIS_LOG_COLOR=false"},
+			validate: func(result string) {
+				g.Expect(result).To(gomega.ContainSubstring("info"))
+				g.Expect(result).To(gomega.ContainSubstring("this is a test"))
+			},
+		},
+		{
+			name: "default",
+			args: []string{"OCIS_LOG_PRETTY=false", "OCIS_LOG_COLOR=false", "OCIS_LOG_LEVEL=error"},
+			validate: func(result string) {
+				g.Expect(result).ToNot(gomega.ContainSubstring("info"))
+				g.Expect(result).ToNot(gomega.ContainSubstring("this is a test"))
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			cmd := exec.Command(os.Args[0], "-test.run=TestDefault")
```

Comment:
Even if it looks strange at first glance, the idea here is not to test an integration but rather a signle unit. in that case that the default logger works as expected.

Since the logger is defined in init, there is no other way (or better, I don't know any other way) how we can ensure that the logger works.

I just looked at how it was solved in the go core code, it's done almost the same way here too.

- https://github.com/golang/go/blob/0784fd1b2f6050b8fb91c88c1d02107258a1651a/src/runtime/vdso_test.go#L52
- https://github.com/golang/go/blob/0784fd1b2f6050b8fb91c88c1d02107258a1651a/src/testing/flag_test.go#L35
- https://github.com/golang/go/blob/0784fd1b2f6050b8fb91c88c1d02107258a1651a/src/os/exec_windows_test.go#L71

Unfortunately, at first it seems a bit strange and unnecessarily difficult, but as I said, I can't think of any other way.

And I don't think it's a good idea to remove the test and continue flying blind.... 
do you have another suggestion?

Reply:
@dragonchaser argued we should get rid of a default logger, after explaining why we all agreed having such a default logger is an anti pattern, gone.